<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ compartmap</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compartmap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.65.87</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/compartmap_vignette.html">A/B compartment inference from ATAC-seq and methylation arrays with compartmap</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/JordanVeldboom/compartmap/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="agrestiCoullCI.html">agrestiCoullCI()</a></code> </p>
        </td>
        <td><p>Agresti-Coull confidence interval for a binomial proportion</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="array.data.chr14.html">meth_array_450k_chr14</a></code> </p>
        </td>
        <td><p>Example Illumina 450k methylation array data for compartmap</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="bootstrapCompartments.html">bootstrapCompartments()</a></code> </p>
        </td>
        <td><p>Non-parametric bootstrapping of compartments and summarization of bootstraps/compute confidence intervals</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="checkAssayType.html">checkAssayType()</a></code> </p>
        </td>
        <td><p>Check if the assay is a SummarizedExperiment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cleanAssayCols.html">cleanAssayCols()</a></code> </p>
        </td>
        <td><p>Remove columns/cells/samples with NAs exceeding a threshold</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cleanAssayRows.html">cleanAssayRows()</a></code> </p>
        </td>
        <td><p>Remove rows with NAs exceeding a threshold</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condenseRE.html">condenseRE()</a></code> </p>
        </td>
        <td><p>Condense a RaggedExperiment to a list of SummarizedExperiments</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condenseSE.html">condenseSE()</a></code> </p>
        </td>
        <td><p>Condense the output of condenseRE to reconstruct per-sample GRanges objects to plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-n_approx.html">.n_approx()</a></code> </p>
        </td>
        <td><p>n_tilde in AC</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-p_approx.html">.p_approx()</a></code> </p>
        </td>
        <td><p>p_tilde in AC</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-z.html">.z()</a></code> </p>
        </td>
        <td><p>Normal alpha/2 quantile</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractOpenClosed.html">extractOpenClosed()</a></code> </p>
        </td>
        <td><p>Get the open and closed compartment calls based on sign of singular values</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fexpit.html">fexpit()</a></code> </p>
        </td>
        <td><p>Helper function: expanded expit</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filterCells.html">filterCells()</a></code> </p>
        </td>
        <td><p>Filter cells using either k-means or Dirichlet process means clustering of sparsity metrics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filterCompartments.html">filterCompartments()</a></code> </p>
        </td>
        <td><p>Filter compartments using confidence estimates and eigenvalue thresholds</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filtered.data.chr14.html">bulkATAC_raw_filtered_chr14</a></code> </p>
        </td>
        <td><p>Example ATAC-seq data for compartmap</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filterOpenSea.html">filterOpenSea()</a></code> <code><a href="filterOpenSea.html">getOpenSeas()</a></code> </p>
        </td>
        <td><p>Filter to open sea CpG loci</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fisherZ.html">fisherZ()</a></code> </p>
        </td>
        <td><p>Fisher's Z transformation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fixCompartments.html">fixCompartments()</a></code> </p>
        </td>
        <td><p>Invert, or "fix", compartments that have a minimum confidence score (1-min.conf)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="flogit.html">flogit()</a></code> </p>
        </td>
        <td><p>Helper function: squeezed logit</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getABSignal.html">getABSignal()</a></code> </p>
        </td>
        <td><p>Calculate Pearson correlations of smoothed eigenvectors</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getArrayABsignal.html">getArrayABsignal()</a></code> </p>
        </td>
        <td><p>Estimate A/B compartments from methylation array data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getAssayNames.html">getAssayNames()</a></code> </p>
        </td>
        <td><p>Get the assay names from a SummarizedExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getATACABsignal.html">getATACABsignal()</a></code> </p>
        </td>
        <td><p>Estimate A/B compartments from ATAC-seq data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getBinMatrix.html">getBinMatrix()</a></code> </p>
        </td>
        <td><p>Generate bins for A/B compartment estimation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getBSABsignal.html">getBSABsignal()</a></code> </p>
        </td>
        <td><p>Estimate A/B compartments from BS-seq data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getChrs.html">getChrs()</a></code> </p>
        </td>
        <td><p>Get the chromosomes from an object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getCorMatrix.html">getCorMatrix()</a></code> </p>
        </td>
        <td><p>Calculate Pearson correlations of a binned matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getGlobalMeans.html">getGlobalMeans()</a></code> </p>
        </td>
        <td><p>Get the global means of a matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getMatrixBlocks.html">getMatrixBlocks()</a></code> </p>
        </td>
        <td><p>Get chunked sets of row-wise or column-wise indices of a matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getSeqLengths.html">getSeqLengths()</a></code> </p>
        </td>
        <td><p>Get the seqlengths of a chromosome</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getShrinkageTargets.html">getShrinkageTargets()</a></code> </p>
        </td>
        <td><p>Get the specified samples to shrink towards instead of the global mean</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getSVD.html">getSVD()</a></code> </p>
        </td>
        <td><p>Compute the SVD of a matrix using irlba</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hdf5TFIDF.html">hdf5TFIDF()</a></code> </p>
        </td>
        <td><p>Transform/normalize compartment calls using TF-IDF on HDF5-backed objects</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ifisherZ.html">ifisherZ()</a></code> </p>
        </td>
        <td><p>Fisher's Z transformation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="imputeKNN.html">imputeKNN()</a></code> </p>
        </td>
        <td><p>Impute missing values/NAs with KNN</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="meanSmoother.html">meanSmoother()</a></code> </p>
        </td>
        <td><p>Windowed mean smoother</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotAB.html">plotAB()</a></code> </p>
        </td>
        <td><p>Plots A/B compartment estimates on a per chromosome basis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="precomputeBootstrapMeans.html">precomputeBootstrapMeans()</a></code> </p>
        </td>
        <td><p>Pre-compute the global means for bootstrapping compartments</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="preprocessArrays.html">preprocessArrays()</a></code> </p>
        </td>
        <td><p>Preprocess arrays for compartment inference</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="preprocessBSseq.html">preprocessBSseq()</a></code> </p>
        </td>
        <td><p>Preprocess BS-seq for compartment inference</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="removeEmptyBoots.html">removeEmptyBoots()</a></code> </p>
        </td>
        <td><p>Remove bootstrap estimates that failed</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scCompartments.html">scCompartments()</a></code> </p>
        </td>
        <td><p>Estimate A/B compartments from ATAC-seq data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="shrinkBins.html">shrinkBins()</a></code> </p>
        </td>
        <td><p>Employ an eBayes shrinkage approach for bin-level estimates for A/B inference</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="smoothBSCounts.html">smoothBSCounts()</a></code> </p>
        </td>
        <td><p>Dirichlet smoothing of BS-seq counts prior to compartment inference</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sparseToDenseMatrix.html">sparseToDenseMatrix()</a></code> </p>
        </td>
        <td><p>Convert a sparse matrix to a dense matrix in a block-wise fashion</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="summarizeBootstraps.html">summarizeBootstraps()</a></code> </p>
        </td>
        <td><p>Summarize the bootstrap compartment estimates and compute Agresti-Coull confidence intervals</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="transformTFIDF.html">transformTFIDF()</a></code> </p>
        </td>
        <td><p>Transform/normalize compartment calls using TF-IDF</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Benjamin Johnson, Tim Triche, Kasper Hansen, Jean-Philippe Fortin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


